src/platform_impl/windows/event_loop.rs

```diff
@@ -853,8 +853,14 @@ unsafe extern "system" fn public_window_callback<T: 'static>(
                     winuser::MonitorFromRect(&new_rect, winuser::MONITOR_DEFAULTTONULL);
                 match fullscreen {
                     Fullscreen::Borderless(ref mut fullscreen_monitor) => {
-                        if new_monitor != fullscreen_monitor.inner.hmonitor()
-                            && new_monitor != ptr::null_mut()
+                        if new_monitor == ptr::null_mut() {
+                            return 0;
+                        }
+
+                        if fullscreen_monitor
+                            .as_ref()
+                            .map(|monitor| new_monitor != monitor.inner.hmonitor())
+                            .unwrap_or(true)
                         {
```

Comment:
Any reason why these aren't combined anymore? Seems strange to have one thing as a gate in the front and the other that does the same thing as a dedicated block.